spec/lib/web_of_science/query_author_spec.rb

```diff
@@ -0,0 +1,121 @@
+# http://savonrb.com/version2/testing.html
+# require the helper module
+require 'savon/mock/spec_helper'
+
+describe WebOfScience::QueryAuthor do
+  include Savon::SpecHelper
+
+  # set Savon in and out of mock mode
+  before(:all) { savon.mock!   }
+  after(:all)  { savon.unmock! }
+
+  subject(:query_author) { described_class.new(author) }
+
+  let(:names) { query_author.send(:names) }
+  let(:institution) { query_author.send(:institution) }
+
+  # These wos_uids are in the wos_search_author_query_response.xml
+  let(:wos_uids) { %w[MEDLINE:29218869 WOS:000422723300003] }
+  let(:wos_search_author_query_response) { File.read('spec/fixtures/wos_client/wos_search_author_query_response.xml') }
+
+  let(:author) do
+    # public data from
+    # - https://stanfordwho.stanford.edu
+    # - https://med.stanford.edu/profiles/russ-altman
+    author = FactoryBot.create(:author,
+                                 preferred_first_name: 'Russ',
+                                 preferred_last_name: 'Altman',
+                                 preferred_middle_name: 'Biagio',
+                                 email: 'Russ.Altman@stanford.edu',
+                                 cap_import_enabled: true)
+    # create some `author.alternative_identities`
+    FactoryBot.create(:author_identity,
+                       author: author,
+                       first_name: 'R',
+                       middle_name: 'B',
+                       last_name: 'Altman',
+                       email: nil,
+                       institution: 'Stanford University')
+    FactoryBot.create(:author_identity,
+                       author: author,
+                       first_name: 'Russ',
+                       middle_name: nil,
+                       last_name: 'Altman',
+                       email: nil,
+                       institution: nil)
+    author
+  end
+
+  it 'works' do
+    expect(query_author).to be_a described_class
```

Comment:
These kind of tests never make any sense.  In Ruby is impossible to call `AnyClass.new` and get back anything but an instance of `AnyClass`.  The only alternative is that initialization throws an exception.  If the intent is to test against that, it should be:
```
expect { described_class.new }.not_to raise_error
```

Otherwise it is just testing "Is Ruby Still Working?"

Reply:
Sort of.  But if the initializer fails by throwing an exception (e.g. arg-exception), then the spec fails.  FYI, I use the rspec-kickstarter gem often to auto-populate specs on a new class and then rework the auto-generated specs a little.  These specs often come from rspec-kickstarter, so if they think it's a useful pattern, that's OK by me too.

Reply:
Right, so if you **intend** to test whether the initializer fails, then I gave you the correct form to use to make that explicit.  Instead you are testing a ruby guarantee and indirectly seeing if it explodes.